js/widgets/musickeyboard.js

```diff
@@ -2624,186 +2619,195 @@ function MusicKeyboard(activity) {
+                                            }
+                                        ],
+                                        0,
+                                        0,
+                                        [thisBlock]
+                                    ]);
+                                }
                             } else {
                                 newStack.push([
-                                    thisBlock + 1,
-                                    [
-                                        "solfege",
-                                        {
-                                            value: SOLFEGECONVERSIONTABLE[note.noteOctave[j][0]]
-                                        }
-                                    ],
+                                    thisBlock,
+                                    "hertz",
                                     0,
                                     0,
-                                    [thisBlock]
+                                    [previousBlock, thisBlock + 1, lastConnection]
                                 ]);
                                 newStack.push([
-                                    thisBlock + 2,
-                                    [
-                                        "number",
-                                        {
-                                            value: note.noteOctave[j][note.noteOctave[j].length - 1]
-                                        }
-                                    ],
+                                    thisBlock + 1,
+                                    ["number", { value: note.noteOctave[j] }],
                                     0,
                                     0,
                                     [thisBlock]
                                 ]);
                             }
-                        } else {
-                            newStack.push([
-                                thisBlock,
-                                "hertz",
-                                0,
-                                0,
-                                [previousBlock, thisBlock + 1, lastConnection]
-                            ]);
-                            newStack.push([
-                                thisBlock + 1,
-                                ["number", { value: note.noteOctave[j] }],
-                                0,
-                                0,
-                                [thisBlock]
-                            ]);
+                            previousBlock = thisBlock;
                         }
-                        previousBlock = thisBlock;
                     }
                 }
             }
-        }
-        this.activity.blocks.loadNewBlocks(newStack);
-        this.activity.textMsg(_("New action block generated!"));
+            setTimeout(this.activity.blocks.loadNewBlocks(newStack), 2000);
```

Comment:
This is heading in the right direction, but not quite correct: each call to loadNewBlocks needs to be a bit further into the future than the previous one, as oppose to all of them the same 2 seconds into the future. So the timeout value needs to be a function of the actionGroup.

Reply:
Also, please see me previous in-line comment about the action name. (With the timeout, we may be able to get away with letting blocks.js assign unique names.

Reply:
Hey @walterbender I will fix this today in updated PR